tests/test_4_immut.py

```diff
@@ -126,6 +126,7 @@ def test_components_vasculature():
     assert_equal(len(morphology.section_types), 3080)
     assert_equal(len(morphology.section(0).predecessors), 0)
     assert_equal(len(morphology.section(0).successors), 2)
+    assert_equal(morphology.section(0).type, 0)
```

Comment:
1) Compare to the actual type, not the int
2) the type is SECTION_NOT_DEFINED, can you change the type in the h5 file to have a more meaningful one.
3) Can you create another test file and put an undefined type (like let's say 12), and create a test with it ?

Reply:
How did you test the neuron type. I just attempted to read a neuron with wrong section type and got the following error message:
```
RawDataError: 
tests/data/simple.swc:25:error
Unsupported section type: 13
```
I think vasculature should print a similar message.

Reply:
Alright.
I checked with the simple.h5 file (changed the type of one of the sections manually)

Reply:
I think this is because the IO files for H5 for both neuron and vasculature do not check the section type at the time of reading - https://github.com/BlueBrain/MorphIO/blob/master/src/plugin/morphologyHDF5.cpp

Reply:
Can you add the check in both files and LBTROW ErrorMessages::ERROR_UNSUPPORTED_SECTION_TYPE ?